util/logutil/general_log_logger.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logutil
+
+import (
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/natefinch/lumberjack"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	generalLogBatchSize = 1024
```

Comment:
Seems that this is no longer a batch size any more? Can it be as large as something like 10000?

Reply:
Yes, since the buffer logic is overloaded to `zapcore.BufferedWriteSyncer`, the original `generalLogBatchSize` is currently meaningless. It's just the maximum pending general log entries.